### PR TITLE
Minor improvements; changed stat format

### PR DESCRIPTION
**Pull request by trag1c**
### Added
- Helper functions for `display_stats`

### Changed
- Made the `Config.load` class method actually use `cls` over `Config`
- Made `Config.save` use `type(self)` over `self.__class__`
- Made `typing` imports and typehints consistent
- Simplified `json.loads(f.read())` to `json.load(f)`
- Simplified `f.write(json.dumps(obj))` to `json.dump(obj, f)`
- Simplified `math.log(n, 2)` to `math.log2(n)`
- Changed exponents to literals in `display_stats`
- `display_stats`' exponents and approximate values will now be floored instead of rounded
  > Example for 1007:
  > `2**10 (1.01K)` -> `2**9.9 (1K)` (`2**10` would actually be displayed once 1024 is hit)
- `display_stats` will now display 2 decimal places instead of 1 for exponents
  > Example for 1000:
  > `2**10 (1K)` -> `2**9.96 (1K)`

### Removed
- Deleted a random comment that did not seem to fit anywhere (?)

### CODE REVIEW

File: mcodingbot/tasks/stats.py [hunk 1]
```diff
@@ -122,27 +121,32 @@ async def get_stats(bot: Bot) -> Stats:
...
+
+
+def floor(number: int | float, ndigits: int = 0) -> float:
+    n: int | float = 10**ndigits
+    return int(number * n) / n
+
+
 def display_stats(stat: int | float) -> str:
-    if stat < 10**3:
+    if stat < 1_000:
```
Comment (Lunarmagpie) [inline mcodingbot/tasks/stats.py:136]: Imo 10**N was more readable here

File: mcodingbot/utils/plugin.py [hunk 2]
```diff
@@ -1,16 +1,16 @@
 from __future__ import annotations
 
-import typing
+from typing import cast, Sequence, TYPE_CHECKING
```
Comment (Lunarmagpie) [inline mcodingbot/utils/plugin.py:3]: Please revert.

File: mcodingbot/tasks/stats.py [hunk 3]
```diff
@@ -122,27 +121,32 @@ async def get_stats(bot: Bot) -> Stats:
     return _last_known_stats
 
 
+def format_float(n: int | float) -> int | float:
+    if float(n).is_integer():
+        return int(n)
+    return n
+
+
+def floor(number: int | float, ndigits: int = 0) -> float:
```
Comment (Lunarmagpie) [inline mcodingbot/tasks/stats.py:130]: This function does not floor. `truncate_decimals` is closer.

File: mcodingbot/utils/__init__.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import typing
+from typing import Sequence
```
Comment (Lunarmagpie) [inline mcodingbot/utils/__init__.py:1]: I dont think there should be a from import here

**Review by Lunarmagpie — CHANGES_REQUESTED** [review]
Mcoding bot uses `import typing` primarily I believe

File: mcodingbot/utils/plugin.py (hunk 2 above)
Comment (trag1c) [reply mcodingbot/utils/plugin.py:3]: Why?

File: mcodingbot/utils/__init__.py (hunk 4 above)
Comment (trag1c) [reply mcodingbot/utils/__init__.py:1]: Why?

File: mcodingbot/tasks/stats.py (hunk 3 above)
Comment (trag1c) [reply mcodingbot/tasks/stats.py:130]: Good point, thanks

File: mcodingbot/utils/__init__.py (hunk 4 above)
Comment (Lunarmagpie) [reply mcodingbot/utils/__init__.py:1]: I thought this project did `import X`

**Comment by trag1c** [comment]
> Mcoding bot uses `import typing` primarily I believe

Before my PR it was 3x `from typing import ...` vs 2x `import typing`

File: mcodingbot/tasks/stats.py [hunk 5]
```diff
@@ -122,27 +121,32 @@ async def get_stats(bot: Bot) -> Stats:
     return _last_known_stats
 
 
+def format_float(n: int | float) -> int | float:
```
Comment (Lunarmagpie) [inline mcodingbot/tasks/stats.py:124]: Would rename this to `strip_trailing_zeros` tbh

**Review by Lunarmagpie — CHANGES_REQUESTED** [review]

File: mcodingbot/tasks/stats.py (hunk 5 above)
Comment (trag1c) [reply mcodingbot/tasks/stats.py:124]: Yeah sounds more accurate, thanks!

File: mcodingbot/tasks/stats.py (hunk 1 above)
Comment (trag1c) [reply mcodingbot/tasks/stats.py:136]: I guess that's subjective; also `1e3` is another option :trollface:

Comment (Lunarmagpie) [reply mcodingbot/tasks/stats.py:136]: I think the relationship between 10**N and 10**N-3 shows the intent more clearly than a different number of 0s

**Comment by CircuitSacul** [comment]
You need to run `isort` locally

**Review by CircuitSacul — APPROVED** [review]
lgtm, I still want to test it but I can't right now

**Review by Lunarmagpie — APPROVED** [review]
its not like 0 or `10**N` matters more than this being merged within the next 50 years

**Pull request merged**